### PR TITLE
[fix] Filter Column when using decorator

### LINKED ISSUE #1188: Decorated grid column prevents use of addFilterColumn()
If you are decorating a grid column with 
`$grid->addDecorator('name', ['Link', ['ticket'], ['tid' => 'id']]);`
the Filter column does not work anymore:
`$grid->addFilterColumn();` 

throws the error "Error : Call to a member function addPopup() on array" on Grid.php:473

### PR DESCRIPTION
**Pull request by ibelar**
Fix: #1188

### CODE REVIEW

**Comment by ibelar** [comment]
@mvorisek - look like #1137 broke cs fixer test. I can push the fix to this pr, but could you please make sure that the suggested changes are ok?

**Comment by mvorisek** [comment]
> 
> 
> @mvorisek - look like #1137 broke cs fixer test. I can push the fix to this pr, but could you please make sure that the suggested changes are ok?

👍 , suggested changes are perfect

File: src/Table.php [hunk 1]
```diff
@@ -326,6 +326,18 @@ public function getColumnDecorators($name)
...
+    /**
+     * Return column instance or first instance if using decorator.
+     *
+     * @param $name
+     *
+     * @return mixed
+     */
+    public function getColumn($name)
+    {
+        return is_array($this->columns[$name]) ? $this->columns[$name][0] : $this->columns[$name];
```
Comment (mvorisek) [inline src/Table.php:337]: is `$this->columns[$name]` guaranteed to have always only one element? If no, the decorator API should be unified. And then this method should be not needed.

**Review by mvorisek — CHANGES_REQUESTED** [review]

File: src/Table.php (hunk 1 above)
Comment (ibelar) [reply src/Table.php:337]: No, it is not guaranteed. When adding decorators, the key will not contain the column instance anymore but an array with column instance set at 0 indexes and the rest as decorators. This is enough for fixing this issue right now. We can work on unifying decorator API in a separate PR.

Comment (mvorisek) [reply src/Table.php:337]: Introducing new methods in a hope to fix them later is bad as it takes much more efford to remove them later.

If this is only hotfix, make the method private and comment this there - and at best create a PR for it and add that link there too.

**Review by DarkSide666 — APPROVED** [review]
Added comment and changed to protected. OK for now.
Probably this new method should also be reused in other methods where we access columns property directly.

**Pull request merged**